### PR TITLE
(DOCSP-14653): Add key-value coding to docs

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Porting the KVC example from Realm.io legacy docs.

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-14653

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-14653/sdk/ios/examples/read-and-write-data/#update-properties-with-key-value-coding

### Review Guidelines
[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by cbush — APPROVED** [review]
Looks good. My only question is whether it would be better as two higher-level headings ("Modify an Object", "Update Properties with Key-Value Coding") so they appear in the "On This Page" box. Up to you.

File: examples/ios/Examples/ReadWriteData.swift [hunk 1]
```diff
@@ -398,6 +398,19 @@ class ReadWriteData: XCTestCase {
...
+
+    func testKeyValueCoding() {
+        // :code-block-start: key-value-coding
+        let realm = try! Realm()
+
+        let allDogs = realm.objects(ReadWriteDataExamples_Dog.self)
+
+        try! realm.write {
+            allDogs.first?.setValue("Sparky", forKey: "name")
+            allDogs.first?.setValue(3, forKey: "age")
```
Comment (ericjordanmossman) [inline examples/ios/Examples/ReadWriteData.swift:410]: Not required, just a thought:
Maybe consider dropping the `first?` from one the lines to demonstrate the "_Avoid the overhead of iterating over a collection while creating accessors for every item._" capability.

```
try! realm.write {
            allDogs.first?.setValue("Sparky", forKey: "name")
            // set each dogs' planet to Earth
            allDogs.setValue("Earth", forKey: "planet")
}
```

The only problem is I couldn't think of any Dog example that all dogs would have in common - other than maybe planet (which I just saw in the old docs).

**Review by ericjordanmossman — APPROVED** [review]
LGTM, I added one comment but not required change.

File: examples/ios/Examples/ReadWriteData.swift (hunk 1 above)
Comment (dacharyc) [reply examples/ios/Examples/ReadWriteData.swift]: Good note. I've updated the example to add a `currentCity`, and then update it for all dogs. Thanks for the feedback!

**Pull request merged**